### PR TITLE
Draft python updates

### PR DESCRIPTION
**Pull request by ethangj**
For review @dennisnewel @mlocher

### CODE REVIEW

File: _basic/builds-and-configuration/dependency-cache.md [hunk 1]
```diff
@@ -30,14 +30,13 @@ Any packages, up to 500mb, in the following directories at the end of your build
...
 - `$HOME/.virtualenv/bin/`
-- `$HOME/.virtualenv/lib/python2.7/site-packages`
 - `$HOME/.m2/repository`
 - `$HOME/.ivy2`
 - `$HOME/cache/yarn`
 - `$HOME/cache/bundler`
 - `$HOME/cache`
 
-Also note that we automatically configure `bundler` to write into `~/cache/bundler` as well as `yarn` to write into `$HOME/cache/yarn`.
+Also note that we automatically configure `bundler` to write into `~/cache/bundler` as well as `yarn` to write into `$HOME/cache/yarn` and `pip` to write into `$HOME/cache`.
```
Comment (mlocher) [inline _basic/builds-and-configuration/dependency-cache.md]: Do we have a specific directory beneath `$HOME/cache` for `pip`?

File: _basic/languages-frameworks/python.md [hunk 2]
```diff
@@ -14,29 +14,21 @@ redirect_from:
 
 ## Versions And Setup
 
-We use **virtualenv** to manage Python environments for you. We currently support Python `2.7.6`. Version `3.4.3` of Python is installed as well and available via the `python3` binary.
+We use **pyenv** to manage Python environments for you.
 
-### Using Python 3
+By default, we run Python version `2.7.13`, but versions `3.4.6`, `3.5.3` and `3.6.1` are all preinstalled as well.
```
Comment (mlocher) [inline _basic/languages-frameworks/python.md]: I would remove the patch levels. Same as with PHP we will do patch level updates and symlink them to the minor level.

File: _basic/languages-frameworks/python.md [hunk 3]
```diff
@@ -14,29 +14,21 @@ redirect_from:
...
 
-The virtual environment from `${HOME}/.virtualenv` is activated and configured for use with Python 2 by default. If you want to switch to version 3, simply add the following commands to your _Setup Commands_.
+All other Python versions can be installed using **pyenv**. If you need the system Python version for any reason, they are available at `/usr/bin/python2` and `/usr/bin/python3`.
```
Comment (mlocher) [inline _basic/languages-frameworks/python.md]: I would remove the references to system Python. We want to get users away from that as much as possible.

I would also add a link to https://github.com/pyenv/pyenv for `pyenv`

File: _basic/languages-frameworks/python.md [hunk 4]
```diff
@@ -14,29 +14,21 @@ redirect_from:
...
 
-If you need a specific Python version installed, use [this script](https://github.com/codeship/scripts/blob/master/languages/python.sh) in your _Setup Commands_.
+You have several options to specify which Python version you would like to use.
 
-First set the desired version as an [environment variable]({{ site.baseurl }}{% link _basic/builds-and-configuration/set-environment-variables.md %}) in your project.
-
-```shell
-PYTHON_VERSION=3.6.0
-```
+In your [setup commands]({{ site.baseurl }}{% link _basic/quickstart/getting-started.md %}) you can use **pyenv** commands. For instance: `pyenv local $version`
```
Comment (mlocher) [inline _basic/languages-frameworks/python.md]: Would use a code block vs inline code

File: _basic/builds-and-configuration/dependency-cache.md (hunk 1 above)
Comment (dennisnewel) [inline _basic/builds-and-configuration/dependency-cache.md]: `$HOME/cache/pip` according to Marko ;)

File: _basic/builds-and-configuration/dependency-cache.md [hunk 5]
```diff
@@ -30,14 +30,14 @@ Any packages, up to 500mb, in the following directories at the end of your build
 - `$REPO_ROOT/node_modules-cache`
 - `$REPO_ROOT/tmp/parallel_runtime_cucumber.log`
 - `$HOME/.virtualenv/bin/`
-- `$HOME/.virtualenv/lib/python2.7/site-packages`
 - `$HOME/.m2/repository`
 - `$HOME/.ivy2`
 - `$HOME/cache/yarn`
 - `$HOME/cache/bundler`
 - `$HOME/cache`
+- `$HOME/cache/pip`
```
Comment (mlocher) [inline _basic/builds-and-configuration/dependency-cache.md]: Not strictly related to the Python PR, but can we sort those folders alphabetically?

File: _basic/builds-and-configuration/dependency-cache.md [hunk 6]
```diff
@@ -30,14 +30,14 @@ Any packages, up to 500mb, in the following directories at the end of your build
 - `$REPO_ROOT/node_modules-cache`
 - `$REPO_ROOT/tmp/parallel_runtime_cucumber.log`
 - `$HOME/.virtualenv/bin/`
```
Comment (mlocher) [inline _basic/builds-and-configuration/dependency-cache.md]: This should also be removed (and any other references to `$HOME/.virtualenv`)

**Comment by dennisnewel** [comment]
sorry - late at the game...there's just a single path that should be updated. The cache directory is $HOME/cache/pip

**Comment by mlocher** [comment]
> sorry - late at the game...there's just a single path that should be updated. The cache directory is $HOME/cache/pip

@dennisnewel Which path references `$HOME/cache` only and should be updated to include the `pip` subfolder?

**Pull request merged**